src/QueryTab/QueryBuilder/FilterSelector.tsx

```diff
@@ -0,0 +1,167 @@
+import { CloseOutlined } from '@ant-design/icons';
+import { Button, Input, InputNumber, List, Select, Typography } from 'antd';
+import { find, findIndex } from 'lodash';
+import React, { FunctionComponent } from 'react';
+import { useT } from '../../shared-react';
+import { ColumnType, TableColumn } from '../../types';
+import { ColumnReference, Filter } from '../types';
+import { CommonProps, removeAt } from './utils';
+
+import './styles.css';
+
+function defaultFilterValue(type: ColumnType) {
+  switch (type) {
+    case 'boolean':
+      return true;
+    case 'real':
+    case 'integer':
+      return 0;
+    case 'text':
+      return '';
+    case 'timestamp':
+      return new Date();
+  }
+}
+
+type FilterValueEditorProps = CommonProps & {
+  column: ColumnReference;
+};
+
+const FilterValueEditor: FunctionComponent<FilterValueEditorProps> = ({ column, query, updateQuery }) => {
+  const t = useT('QueryTab::QueryBuilder::FilterSelector::FilterValueEditor');
+
+  const filter = find(query.filters, { column });
+
+  if (!filter) {
+    return null; // Should never happen.
```

Comment:
maybe a warning to console, just in case?

Reply:
I intentionally avoided prints in renders. In all other actions it prints a warning. It shouldn't happen either way. We can even throw here.